libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -156,7 +156,13 @@ bool SyntaxChecker::visit(PragmaDirective const& _pragma)
 		vector<Token> tokens(_pragma.tokens().begin() + 1, _pragma.tokens().end());
 		vector<string> literals(_pragma.literals().begin() + 1, _pragma.literals().end());
 		SemVerMatchExpressionParser parser(tokens, literals);
-		auto matchExpression = parser.parse();
+		auto matchExpressionOrError = parser.parse();
+		if (std::holds_alternative<SemVerError>(matchExpressionOrError))
+		{
+			auto matchExpressionError = std::get<SemVerError>(matchExpressionOrError);
+			BOOST_THROW_EXCEPTION(matchExpressionError);
```

Comment:
This shouldn't directly throw exception but rather use `m_errorReporter.syntaxError` instead.

Reply:
Oh and we certainly need a source location here!

Reply:
The one thing i wasn't sure about here - would it technically be a `fatalError` in which case we should have a `fatalSyntaxError` which we would throw? Or the `syntaxError` be enough where we would add to `m_errorList` and only throw if there are excessive errors?

Reply:
Actually I think the parser ("parsePragmaVersion") already generates a fatal error if there is anything wrong. This means at this point here, there should not be any error and you could use `solAssert` to check that.